app/client/ui/userTrustsCustomWidget.ts

```diff
@@ -0,0 +1,78 @@
+import {Computed, dom, makeTestId, Observable, styled} from 'grainjs';
+import {makeT} from 'app/client/lib/localization';
+import {getConfirmText, saveModal} from 'app/client/ui2018/modals';
+import {alert} from 'app/client/ui2018/alerts';
+import {labeledSquareCheckbox} from 'app/client/ui2018/checkbox';
+import {inlineMarkdown} from 'app/client/lib/markdown';
+import {gristThemeObs} from 'app/client/ui2018/theme';
+
+const testId = makeTestId('test-custom-widget-warning-modal-');
+
+/**
+ * Show a modal to the user asking him to confirm he trusts the custom widget he's about to install.
+ *
+ * @returns Promise<boolean> Promise that resolves to true if the user confirms he trusts the widget, false otherwise.
+ */
+export function userTrustsCustomWidget() {
+  const t = makeT('userTrustsCustomWidget');
+  const src = gristThemeObs().get().appearance === 'light'
+    ? 'img/security-alert.png'
+    : 'img/security-alert-dark-theme.png';
+
+  let setUserChoice: (choice: boolean) => any;
+  const waitForUserChoice = new Promise<boolean>((resolve) => {
+    setUserChoice = resolve;
+  });
+
+  saveModal((ctl, owner) => {
+    const confirmIsChecked = Observable.create(owner, false);
+    return {
+      title: t('Be careful with unknown custom widgets'),
+      body: dom('div',
+        cssImage(
+          dom('img', {src, alt: '', width: '100', height: '100'}),
```

Comment:
There is no alt for this image, but maybe this is acceptable here?

Reply:
Thanks for the explanation!

Reply:
You may resolve the discussion here, I don't have the rights for that (which is a bit weird)